spec/ruby/core/io/close_on_exec_spec.rb

```diff
@@ -45,8 +45,22 @@
       -> { @io.close_on_exec = true }.should raise_error(IOError)
     end
 
-    it "returns nil" do
-      @io.send(:close_on_exec=, true).should be_nil
+    it "returns given argument" do
+      truthy = Object.new
+      (@io.close_on_exec = truthy).should equal(truthy)
+    end
+
+    ruby_version_is ''...'3.1' do
+      it "returns nil when called with send" do
+        @io.send(:close_on_exec=, true).should be_nil
+      end
+    end
```

Comment:
I think we could also just remove this spec because it doesn't make much sense to return `nil` there.

Reply:
Sure! Removed in https://github.com/ruby/ruby/pull/4321/commits/d3ca24906ac0e1fe706dd4ef8be50556c3ec0fc9 🚮